python/hail/dataset.py

```diff
@@ -750,6 +750,63 @@ def annotate_variants_expr(self, condition):
         pargs = ['annotatevariants', 'expr', '-c', condition]
         return self.hc.run_command(self, pargs)
 
+    def annotate_variants_keytable(self, keytable, condition, vdsKey=None):
```

Comment:
Ok, I suggest the API be:

 - `vds_key is None` (default): use efficient join algorithm on `Variant`
 - `vds_key is list of hail-expressions`: for each variant, run the list of hail-expressions, and combine into a `Row` to use as the join key with the `KeyTable`